hivemind/averaging/key_manager.py

```diff
@@ -93,51 +78,16 @@ async def get_averagers(self, group_key: GroupKey, only_active: bool) -> List[Tu
         if result is None or not isinstance(result.value, dict):
             logger.debug(f"Allreduce group not found: {group_key}, creating new group.")
             return []
-        averagers = [
-            (PeerID(key), looking_for_group.expiration_time)
-            for key, looking_for_group in result.value.items()
-            if key != self.RESERVED_KEY_FOR_NBITS and (not only_active or looking_for_group.value)
-        ]
-        num_active_averagers = sum(
-            1
-            for key, looking_for_group in result.value.items()
-            if key != self.RESERVED_KEY_FOR_NBITS and looking_for_group.value
-        )
-
-        suggested_nbits = self.get_suggested_nbits(result)
-        if (
-            suggested_nbits is not None
-            and suggested_nbits != len(self.group_bits)
-            and suggested_nbits != self.suggested_nbits
-        ):
-            self.suggested_nbits = suggested_nbits
-            logger.warning(f"{self.peer_id} - another averager suggested {self.suggested_nbits}-bit keys")
-        elif num_active_averagers >= self.excessive_size:
-            self.suggested_nbits = max(suggested_nbits or 0, len(self.group_bits) + 1)
-            logger.warning(f"{self.peer_id} - too many peers in bucket, switching to {self.suggested_nbits}-bit keys")
+        averagers = []
+        for key, entry in result.value.items():
+            try:
+                if only_active and not entry.value:
+                    continue
+                averagers.append((PeerID(key), entry.expiration_time))
+            except Exception as e:
+                logger.warning(f"Could not parse group key {key} ({entry}, exc={e})")
```

Comment:
logger.exception? Unparseable keys are not a sign of normal operation

Reply:
Alas it isn't: it may happen if some peers are messing around, i would like to insist in logs that this is not critical for us